src/DiffWindow.ts

```diff
@@ -39,32 +37,21 @@ export class DiffWindow {
 				title: 'Cancel'
 			}
 		]);
-
-        this.noteSelectWindow = new NoteSelectWindow(this.joplinDialogs, this.joplinData, this.fileSystem, this.joplinInstallDir);
-        await this.noteSelectWindow.init("dialog-note-select");
     }
 
-    public async OpenWindow(noteId: string, compareWithId : string = "") {
+    /**
+     * Opens the diff window between 2 different notes and returns the merged result as a string.
+     * 
+     * @param noteId The ID of the base note which will be saved eventually.
+     * @param compareWithId The ID of the note that we should compare to.
+     * @returns `null` if the merge was cancelled. Otherwise, the contents of the merged notes. 
+     */
+    public async OpenWindow(noteId: string, compareWithId : string) : Promise<DiffResult> {
```

Comment:
Should return `Promise<DiffResult | null>`. Adding "null" is good because it can help detect errors in your code, where you don't do a null check.

Reply:
Added.